components/renku_data_services/storage/rclone.py

```diff
@@ -156,6 +157,91 @@ def __patch_schema_remove_oauth_propeties(spec: list[dict[str, Any]]) -> None:
                         options.append(option)
                 storage["Options"] = options
 
+    @staticmethod
+    def __find_webdav_storage(spec: list[dict[str, Any]]) -> dict[str, Any]:
+        """Find and return the WebDAV storage schema from the spec."""
+        webdav_storage = next((s for s in spec if s["Prefix"] == "webdav"), None)
+        if not webdav_storage:
+            raise errors.ValidationError(message="WebDAV storage not found in schema.")
+        return deepcopy(webdav_storage)
+
+    @staticmethod
+    def __add_webdav_based_storage(
+        spec: list[dict[str, Any]],
+        prefix: str,
+        name: str,
+        description: str,
+        url_value: str,
+        public_link_help: str,
+    ) -> None:
+        """Create a modified copy of WebDAV storage and add it to the schema."""
+        # Find WebDAV storage schema and create a modified copy
+        storage_copy = RCloneValidator.__find_webdav_storage(spec)
+        storage_copy.update({"Prefix": prefix, "Name": name, "Description": description})
+
+        custom_options = [
+            {
+                "Name": "access",
```

Comment:
question: I'm a bit confused by this, this is a field called access, but lower down its values are used as provider. In rclone schemas, the providers should be in the field `provider` by convention.

Reply:
Oh, I see where the confusion came from. I was trying to make it flexible by adding an option for access (personal and shared), but it ended up being a bit unclear. I’ll change it so the modes are set in the provider options instead, and I’ll use the provider property to specify when an option applies to a specific mode (personal or shared).